enhancements/installer/component-selection.md

```diff
@@ -0,0 +1,432 @@
+This proposal consists of two parts:
+
+1) Formalizing a concept of a "capability" annotation which allows a given resource to be excluded based
+on installer input. For example the console related resources could be annotated as
+
+```yaml
+annotations:
+  capability.openshift.io/console: "true"
+```
+
+Note: "true" means "included by default unless explicitly excluded", in the future we may enable capabilities
+to mark themselves as "false" meaning "excluded by default, unless explicitly included".
+
+
+2) Defining an [install config api](https://github.com/openshift/installer/blob/790048166067273d34f76bea4220fa395b1cce1b/pkg/types/installconfig.go#L70) field whereby the user can opt out of specific capabilities.
+
+```yaml
+InstallConfig.Capabilities
+  Default: [include|exclude]
+  Exclude:
+  - console
+  - samples
+  Include:
+  - registry
+```
+
+Which resources ultimately get installed for a given cluster would be the set of resources encompassed
+by the CLUSTER_PROFILE(if any), minus any resources explicitly excluded by the excluded capabilities configuration,
+plus any explicitly included capabilities.  Any resource that is not included by the CLUSTER_PROFILE, is not
+included via the Include stanza, and is not Excluded via the Exclude stanza, will be filtered or not based on
+the `Default` value (include or exclude).
+
+The `Default` field (carried through to the CVO) allows an Admin to choose what behavior they want for new
+capabilities that might come as part of a future upgrade (either have the new capability installed during
+the upgrade, or exclude it)
+
+Specifying the same capability as an `include` and an `exclude` is an error condition.
+
+
+Examples of candidate components to be treated as capabilities:
+
+* console
+* imageregistry
+* samples
+* cluster baremetal operator
+* olm/marketplace
+* csi-*
+* insights
+* monitoring
+* ???
+
+
+
+3) Pass the list of filtered annotations to the CVO.  This is distinct from overrides because overrides
+put the cluster in an unsupported state.  Filtered annotations are supported for upgrades.  The filtered
+components will be listed in the [ClusterVersion](https://github.com/openshift/api/blob/4436dc8be01e8dcd8b250e1b32bb0fbd64ba78ac/config/v1/types_cluster_version.go#L35) object:
+
+```yaml
+  spec:
+    capabilities:
+      default: [include|exclude]
+      exclude:
+      - console
+      - samples
+      include:
+      - registry
+```
+
+The CVO will filter out(not apply/reconcile) resources that are annotated with `capability.openshift.io/$exclusions`
+and listed in the `exclude` section, or are not listed and the `default` setting is `exclude`.  In addition
+any explicitly included capability should be added, regardless of the CLUSTER_PROFILE.
+
+If a resource has multiple `capability.openshift.io` annotations, then the resource will be filtered if any
```

Comment:
The justification for this choice is described further down.  (Namely, the interaction between the Monitoring capability(provides the servicemonitor CRD) and all the capabilities that include service monitor definitions).  That struck me as a strong enough use case/scenario to choose the behavior described.

I agree it may create some challenges, the reality is that either choice is going to have challenges, but I can't think of an example scenario where a single resource would be defined as being part of 2 different capabilities *and* you would want that resource to be instantiated if *either* of those capabilities were enabled, not only when both of them are.  Do you have one?  In general i would not expect cases of a single resource belonging to multiple capabilities.  The ServiceMonitor one is really more of a dependency relationship.